### ISSUE TITLE
Query Task Records with pagination and filter like status filter

### ISSUE THREAD

**Issue by LouisAldorio** [781flyingdutchman/background_downloader#109]
**Is your feature request related to a problem? Please describe.**
The only option available now is taking all record from the database or from certain timestamp, it will be very useful to actually have pagination option and filtering option such as filter by status

**Comment by 781flyingdutchman** [comment]
Hi again!  This is quite easy to do yourself once you have the list of TaskRecords, eg: 
```
for (var task in taskRecords.where((record) => record.status == TaskStatus.complete)) 
  { 
    // record.task is a completed task, so do something with it 
  }
```
It's hard to predict the filtering developers will want - besides filtering on status, what other filters do you think would be helpful?

Also, if you want more control you can also use the `SqlitePersistentStorage` approach and keep a reference to the database, which has a typical `where: , whereargs: ` structure.

**Comment by 781flyingdutchman** [comment]
Included `allRecordsWithStatus` in V7.9.0

**Comment by LouisAldorio** [comment]
Okay thankyou sir
